great_expectations/rule_based_profiler/parameter_builder/regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,213 @@
+        """
+        metric_computation_result: MetricComputationResult
+
+        metric_values: np.ndarray
+
+        metric_computation_result: MetricComputationResult = self.get_metrics(
+            metric_name="column_values.nonnull.count",
+            metric_domain_kwargs=self.metric_domain_kwargs,
+            metric_value_kwargs=self.metric_value_kwargs,
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+        metric_values = metric_computation_result.metric_values
+        # Now obtain 1-dimensional vector of values of computed metric (each element corresponds to a Batch ID).
+        metric_values = metric_values[:, 0]
+
+        nonnull_count: int = sum(metric_values)
+
+        regex_string_success_ratios: dict = {}
+
+        # Obtain candidate_regexes from "rule state" (i.e, variables and parameters); from instance variable otherwise.
+        candidate_regexes: Union[
+            Set[str],
+            List[str],
+            "RegexPatternStringParameterBuilder.CANDIDATE_REGEX",  # noqa: F821
+        ] = get_parameter_value_and_validate_return_type(
+            domain=domain,
+            parameter_reference=self.candidate_regexes,
+            expected_return_type=None,
+            variables=variables,
+            parameters=parameters,
+        )
+        if candidate_regexes is not None and isinstance(candidate_regexes, list):
+            candidate_regexes = set(candidate_regexes)
+        else:
+            candidate_regexes = RegexPatternStringParameterBuilder.CANDIDATE_REGEX
+
+        regex_string: str
+        match_regex_metric_value_kwargs: dict
+        for regex_string in candidate_regexes:
+            if self.metric_value_kwargs:
+                match_regex_metric_value_kwargs: dict = {
+                    **self._metric_value_kwargs,
+                    **{"regex": regex_string},
+                }
+            else:
+                match_regex_metric_value_kwargs: dict = {"regex": regex_string}
+
+            metric_computation_result: MetricComputationResult = self.get_metrics(
+                metric_name="column_values.match_regex.unexpected_count",
+                metric_domain_kwargs=self.metric_domain_kwargs,
+                metric_value_kwargs=match_regex_metric_value_kwargs,
+                domain=domain,
+                variables=variables,
+                parameters=parameters,
+            )
+            metric_values = metric_computation_result.metric_values
+            # Now obtain 1-dimensional vector of values of computed metric (each element corresponds to a Batch ID).
+
+            metric_values = metric_values[:, 0]
+            match_regex_unexpected_count: int = sum(metric_values)
+            success_ratio: float = (
+                nonnull_count - match_regex_unexpected_count
+            ) / nonnull_count
+            regex_string_success_ratios[regex_string] = success_ratio
```

Comment:
As with all parameterbuilders, we need to be sure we're not computing metrics one-at-a-time like this. Get the list of all the metrics you want, then ask the validator for all of them at the same time.

Reply:
Can you explain that change more? I'm not seeing anything here that *could* require an upstream change

Reply:
It would be a change in `def get_metrics()` in the `ParameterBuilders` that calls `get_metrics()` in `Validator`.  It requires building and parsing a `dict` (currently a single value is passed in) that will be passed from the ParameterBuilder to the Validator. 

Since this is a change at the base `ParameterBuilders`, it is something that would affect to both Regex and SimpleDataFormatString ParameterBuilders.

Reply:
@alexsherstinsky thank you for the synchronous discussion on this